compiler/src/dotty/tools/dotc/typer/QuotesAndSplices.scala

```diff
@@ -72,7 +72,7 @@ trait QuotesAndSplices {
         ctx.warning("Canceled quote directly inside a splice. ${ '{ XYZ } } is equivalent to XYZ.", tree.sourcePos)
       case _ =>
     }
-    if (ctx.mode.is(Mode.QuotedPattern) && level == 1)
+    if (ctx.mode.is(Mode.QuotedPattern))
```

Comment:
Same question as above: the check becomes invalid or redundant?

Reply:
Same reason